mobile/native/src/ln_dlc/mod.rs

```diff
@@ -224,6 +245,28 @@ pub fn run(data_dir: String, seed_dir: String, runtime: &Runtime) -> Result<()>
         let _running = node.start(event_handler)?;
         let node = Arc::new(Node::new(node, _running));
 
+        std::thread::spawn({
+            let node = node.clone();
+            move || {
+                // We wait a minute to not interfere with the app's startup. We want to be able to
+                // quickly refresh the wallet _before_ the first on-chain sync, as this one can take
+                // a long time and blocks the wallet refresh since they both need to acquire the
+                // same mutex.
+                //
+                // TODO: This should not be necessary as soon as we rewrite the on-chain wallet with
+                // bdk:1.0.0.
+                std::thread::sleep(Duration::from_secs(60));
```

Comment:
🤔 calling the `keep_wallet_balance_and_history_up_to_date` function first, may prevent us from having to sleep here?

Reply:
I like your suggestion! I'll give it a go.